visitor.go

```diff
@@ -249,3 +256,106 @@ func (v *treeVisitor) isSupported(tk token.Token) bool {
+func (v *treeVisitor) evaluateConstExpr(expr ast.Expr) string {
+	// Handle binary expressions like Prefix + "suffix"
+	if binExpr, ok := expr.(*ast.BinaryExpr); ok && binExpr.Op == token.ADD {
+		// We're only interested in string concatenation
+		leftVal := v.resolveExprToString(binExpr.X)
+		rightVal := v.resolveExprToString(binExpr.Y)
+		
+		// If both sides resolved to strings, combine them
+		if leftVal != "" && rightVal != "" {
+			return leftVal + rightVal
+		}
+	} else {
+		// Handle single identifiers (could be constants)
+		return v.resolveExprToString(expr)
+	}
+	
+	return ""
+}
+
+// resolveExprToString tries to resolve an expression to its string value.
+// Handles identifiers (looking up constants), string literals, and nested expressions.
+func (v *treeVisitor) resolveExprToString(expr ast.Expr) string {
+	switch e := expr.(type) {
+	case *ast.BasicLit:
+		// Direct string literal
+		if e.Kind == token.STRING {
+			val, err := strconv.Unquote(e.Value)
+			if err == nil {
+				return val
+			}
+			// Fall back to striping quotes manually if unquoting fails
+			if len(e.Value) >= 2 {
+				return e.Value[1 : len(e.Value)-1]
+			}
+		}
+		
+	case *ast.Ident:
+		// Reference to a constant
+		// Check if we've already seen this constant in the current package
+		v.p.constMutex.RLock()
+		defer v.p.constMutex.RUnlock()
+		
+		for val, constList := range v.p.consts {
+			for _, c := range constList {
+				// Match by name and package
+				if c.Name == e.Name && c.packageName == v.packageName {
+					return val
+				}
+			}
+		}
+		
+	case *ast.BinaryExpr:
+		// Recursively evaluate nested expressions
+		if e.Op == token.ADD {
+			left := v.resolveExprToString(e.X)
+			right := v.resolveExprToString(e.Y)
+			if left != "" && right != "" {
+				return left + right
+			}
+		}
+	
+	case *ast.ParenExpr:
+		// Handle parenthesized expressions
+		return v.resolveExprToString(e.X)
+	}
+	
+	return ""
+}
+
+// addConstWithValue adds a constant with an already evaluated string value.
+// This is similar to addConst but skips the unquoting step since the value is already processed.
+func (v *treeVisitor) addConstWithValue(name string, val string, pos token.Pos) {
+	// Skip constants with values that would be filtered anyway
+	if len(val) < v.p.minLength {
+		return
+	}
+
+	if v.ignoreRegex != nil && v.ignoreRegex.MatchString(val) {
+		return
+	}
+
+	// Use interned string to reduce memory usage
+	internedVal := InternString(val)
+	internedName := InternString(name)
+	internedPkg := InternString(v.packageName)
+
+	// Lock to safely update the shared map
+	v.p.constMutex.Lock()
+	defer v.p.constMutex.Unlock()
+
+	// track this const if this is a new const, or if we are searching for duplicate consts
+	if _, ok := v.p.consts[internedVal]; !ok || v.p.findDuplicates {
+		v.p.consts[internedVal] = append(v.p.consts[internedVal], ConstType{
+			Name:        internedName,
+			packageName: internedPkg,
+			Position:    v.fileSet.Position(pos),
+		})
+	}
+}
```

Comment:
Parsing the AST might be preferable to the type checking approach for strings. We should probably benchmark the two and pick whichever is faster in that case.

I'd guess most of the overhead would come from running the type-checker itself. It seems `golangci-lint` will precompute type info whether we need it or not, so it's not clear how much a speed difference will matter in that case.

Parsing constant expressions with numbers could also be implemented manually, but for numeric types, I'd expect maintaining compatibility with what Golang would compute would probably be more difficult.

Reply:
I prefer your approach, using the typechecker might impact slightly the performance but I just recently optimized the processing and gained a bit over 2x and I doubt here we're losing as much.